src/NetscriptFunctions/Corporation.ts

```diff
@@ -415,7 +416,7 @@ export function NetscriptCorporation(): InternalAPI<NSCorporation> {
       const option = helpers.string(ctx, "option", _option);
       if (!hasUnlock(CorpUnlockName.SmartSupply))
         throw helpers.makeRuntimeErrorMsg(ctx, `You have not purchased the Smart Supply upgrade!`);
-      SetSmartSupplyOption(warehouse, material, option);
+      SetSmartSupplyOption(warehouse, material, option as CorpSmartSupplyOption);
```

Comment:
So here you're passing option to a function that requires a valid CorpSmartSupplyOption, but you're not doing any type validation on the input other than that it is a string. Either the function is meant to handle unvalidated strings (and should take in a string for that parameter) or it is meant to take in validated strings (and validation should be added before calling the function). Related to other comment

Reply:
This is something I considered as well. I had to choose between this function accepting any string as an argument in-game and having validation inside or accepting a strict set of strings but still having validation inside (just in case players call it with a wrong value). I found the second option more appealing

Reply:
sorry, the comment above was actually for `Actions.ts` file

Reply:
What we can do is - move assertion part from `Actions.ts` file here since the in-game code will not call `SetSmartSupplyOption` with an invalid option.
wdyt?

Reply:
We validate player input using things like checkEnum, which also works for typed string arrays (not just enums). The other PR I'm working on will replace checkEnum with something similar that also handles making the error message, checkEnum just returns a boolean for whether the input was the right type or not.

You can search NetscriptFunctions/Corporation.ts for `checkEnum` and see what the normal usage is for type guarding player input with checkEnum. By the time we are calling an ingame function that has required types, we always should have already done the type validation.

Reply:
oh, I've already updated this solution with `assertMember`.